src/fileseq/utils.py

```diff
@@ -388,19 +391,20 @@ def pad(number, width=0, decimal_places=None):
     Return the zero-padded string of a given number.
 
     Args:
-        number (int, float, or decimal.Decimal): the number to pad
+        number (str, int, float, or decimal.Decimal): the number to pad
         width (int): width for zero padding the integral component
         decimal_places (int): number of decimal places to use in frame range
 
     Returns:
         str:
     """
-
     # Make the common case fast. Truncate to integer value as USD does.
     # https://graphics.pixar.com/usd/docs/api/_usd__page__value_clips.html
     # See _DeriveClipTimeString for formating of templateAssetPath
     # https://github.com/PixarAnimationStudios/USD/blob/release/pxr/usd/usd/clipSetDefinition.cpp
     if decimal_places == 0:
+        if not isinstance(number, str):
+            number = round(number)
```

Comment:
We have to be careful about being able to handle different string types, so you will see other places in code test `isinstance(number, futils.string_types)`. What if we just go with the safer approach and try/catch the round
```python
try:
    number = round(number)
except TypeError:
    pass
```

Reply:
Sounds good, thanks for the snippet!